scripts/get_docker_info.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/sh
+
+version_only=
+if [ "$1" = "-v" ]; then
+  version_only=1
+  shift 1
+fi
+
+dockerfile=$1
+shift 1
+
+if [ ! -f  $dockerfile ]; then
+  echo "Dockerfile not found" >> /dev/stderr
+  exit 1
+fi
+
+if [ -z "$1"  ]; then
+  echo "No target specified" >> /dev/stderr
+  exit 1
+fi
+
+tag=$(grep "AS $1\$" $dockerfile | sed -e 's/FROM *//' -e 's/ *AS .*//')
+if [ -z "$tag" ]; then
+  echo "Target $1 not defined" >> /dev/stderr
+  exit 1
+fi
+if [ "$version_only" = "1" ]; then
+   echo $tag | sed -e 's/.*://'
+else
+   echo $tag
+fi
```

Comment:
😳 Could we write this in Python, pretty please? 😂 

Something like

```python
import argparse
import re

from_line_re = re.compile(r"FROM\s+(?P<image>\S+)\s+AS\s+(?P<target>\S+)")


def find_image_in_dockerfile(dockerfile, target):
    with open(dockerfile) as f:
        for line in f.readlines():
            if (m := from_line_re.match(line)) and m.group("target") == target:
                return m.group("image")
    raise ValueError(f"Target {target} not defined in {dockerfile}")


def main():
    ap = argparse.ArgumentParser()
    ap.add_argument("dockerfile", required=True)
    ap.add_argument("target", required=True)
    ap.add_argument("-v", "--version-only", action="store_true")
    args = ap.parse_args()
    image = find_image_in_dockerfile(args.dockerfile, args.target)
    if args.version_only:
        image = image.rpartition(":")[-1]
    print(image)
```
, I think...

Reply:
No objection from me 😃   I just took this script off my library of useful stuff TBH